src/common/mpi.jl

```diff
@@ -0,0 +1,49 @@
+# Convenience functions for working with MPI
+using MPI
+
+"""
+Initialize MPI. Must be called before doing any non-trivial MPI work
+(even in the single-process case). Unlike the MPI.Init() function,
+this can be called multiple times.
+"""
+function mpi_ensure_initialized()
+    # TODO look more closely at interaction between MPI and threads
+    MPI.Initialized() || MPI.Init_thread(MPI.ThreadLevel(3))
+end
+
+"""
+Number of processors used in MPI. Can be called without ensuring initialization.
+"""
+mpi_nprocs() = mpi_nprocs(MPI.COMM_WORLD)
+mpi_nprocs(comm) = (mpi_ensure_initialized(); MPI.Comm_size(comm))
```

Comment:
We have underscores after the `n`, so `mpi_n_procs`?

Reply:
Depends; eg julia has nthreads. I think I like `nprocs` better

Reply:
yes but we use the convention `n_` in many places. (n_iter, n_spin_components, n_electrons, n_k, ...)

Reply:
Also julia has set_num_threads ... so it's not perfectly consistent either. It's just driving me mad to rember where it is which way. At least in our code I would like consistency.

Reply:
These things are always going to be inconsistent, for good reason: naming things is hard, if you impose consistency it becomes impossible. I would just relax and accept it :-) For instance, my preferred spelling would be `niter`, `n_spin_components`, `n_electrons`, `nk`, etc. Readability trumps consistency. In this case I'd sync nprocs on nthreads

Reply:
ok that's a good point. Fair enough ... I give in!